### ISSUE TITLE
Refactoring namespaces

### ISSUE THREAD

**Issue by Taluu** [phpactor/phpactor#275]
Hey,

As the `phpactor references:class ...` supports changing classes names, the equivalent to change a namespace name would be awesome. 

Is some support planned ?

BTW, not sure that `references:class ... --replace` is really good for UX ; AFAIC, before digging into the readme, and just watching the command list, I did not expect that this command would be used to rename classes.

Cheers :}

**Comment by dantleech** [comment]
Well, you can already do `class:move lib/Foobar lib/Barfoo` (or `class:move "Acme\\Foobar" "Acme\\Barfoo`). But I guess this doesn't work for references.

Planning also to allow moving by [glob/regex](https://github.com/phpactor/phpactor/issues/263).

For the UX, not sure, the `class:move` and `references:class|member` commands could probably be merged somehow, and then there could just be 2 operations (find / replace).

**Comment by Taluu** [comment]
Yup, it works for folders and classes, but not namespaces (e.g `class:move Acme AcmeFoo`, so that all `Acme\Foo` are moved to `AcmeFoo\Foo`). Glob/regexes would be awesome for that. :}

For the ux, I would have expected something along `refactor` or something. the `references:class|member` should be really just finding the references instead of having a `--replace` flag, that was my point. :}

**Comment by jjanvier** [comment]
Yep, same need here.
Do you plan to do it? Can I help somehow?

**Comment by dantleech** [comment]
Yeah, this is quite high on my list because it would be _awesome_ to refactor class names by regex.

**Comment by kermorgant** [comment]
In a similar way, updating namespace after moving a class from one namespace to another would be nice.

Example : During  a migration from sf3 to sf4, move `src/AppBundle/AppEvents.php` from `AppBundle` namespace to `src/AppEvents.php` in `App` namespace.

**Comment by dantleech** [comment]
> In a similar way, updating namespace after moving a class from one namespace to another would be nice.

This should already work?

```
$ phpactor references:class "src/AppBundle/AppEvents.php" --replace="src/AppEvents.php"
```

Although, as this is a project file, it seems `class:move` would have been better to start with?

**Comment by kermorgant** [comment]
Didn't know I could handle it that way, so I only tried class:move but the namespace was not updated after the move.
